app/javascript/geoblacklight/metadata_download.js

```diff
@@ -0,0 +1,48 @@
+export class GeoBlacklightMetadataDownloadButton {
+  constructor(el, options = {}) {
+    this.options = options;
+    this.el = typeof el === "string" ? document.querySelector(el) : el;
+    this.download = document.querySelector(
+      this.options.target || this.el.getAttribute("data-ref-download")
+    );
+    // On initialization only do this for the first one.
+    if (i === 0) {
+      this.setRefUrl();
+    }
+    this.configureHandler();
+  }
+
+  configureHandler() {
+    this.el.addEventListener("click", () => this.setRefUrl());
+  }
+
+  setRefUrl() {
+    const refUrl = this.el.getAttribute("data-ref-endpoint");
+    if (!refUrl) {
+      this.download.style.display = "none";
+    } else {
+      this.download.style.display = "";
+      this.download.setAttribute("href", refUrl);
+    }
+  }
+}
+
+export default function initializeMetadataDownload() {
```

Comment:
Metadata download is working locally. I am guessing it has something to do with the fact this function used to be in modules and had a wrapper Blacklight.onLoad

https://github.com/geoblacklight/geoblacklight/blob/24aba2e82f4b386e0301d8beebf969727bcf7fb0/app/assets/javascripts/geoblacklight/modules/metadata.js#L3